crrExistingObjects.js

```diff
@@ -0,0 +1,166 @@
+    log.fatal('SECRET_KEY not defined');
+    process.exit(1);
+}
+const options = {
+    accessKeyId: ACCESS_KEY,
+    secretAccessKey: SECRET_KEY,
+    endpoint: ENDPOINT,
+    region: 'us-east-1',
+    sslEnabled: false,
+    s3ForcePathStyle: true,
+    apiVersions: { s3: '2006-03-01' },
+    signatureVersion: 'v4',
+    signatureCache: false,
+    httpOptions: {
+        maxRetries: 0,
+        timeout: 0,
+        agent: new http.Agent({ keepAlive: true }),
+    },
+};
+const s3 = new AWS.S3(options);
+
+
+const bb = new BackbeatClient(options);
+
+function _markObjectPending(bucket, key, versionId, repConfig, cb) {
+    return waterfall([
+        // get object blob
+        next => bb.getMetadata({
+            Bucket: bucket,
+            Key: key,
+            VersionId: versionId,
+        }, next),
+        // update replication info and put back object blob
+        (mdRes, next) => {
+            const objMD = JSON.parse(mdRes.Body);
+            if (objMD.replicationInfo && objMD.replicationInfo.status !== '') {
+                // skip object since it's already marked for crr
+                return next();
+            }
+            const { Rules, Role } = repConfig;
+            const destination = Rules[0].Destination.Bucket;
+            const storageClass = Rules[0].Destination.StorageClass;
+            // set replication properties
+            const ops = objMD['content-length'] === 0 ? ['METADATA'] :
+                ['METADATA', 'DATA'];
+            const backends = [{
+                site: storageClass,
+                status: 'PENDING',
+                dataStoreVersionId: '',
+            }];
+            const replicationInfo = {
+                status: 'PENDING',
+                backends,
+                content: ops,
+                destination,
+                storageClass,
+                role: Role,
+                storageType: '',
+            };
+            objMD.replicationInfo = replicationInfo;
+            const mdBlob = JSON.stringify(objMD);
+            return bb.putMetadata({
+                Bucket: bucket,
+                Key: key,
+                ContentLength: Buffer.byteLength(mdBlob),
+                Body: mdBlob,
+            }, next);
+        },
+    ], cb);
+}
+
+// list object versions
+function _listObjectVersions(bucket, VersionIdMarker, KeyMarker, cb) {
+    return s3.listObjectVersions({
+        Bucket: bucket,
+        MaxKeys: LISTING_LIMIT,
+        VersionIdMarker,
+        KeyMarker,
+    }, cb);
+}
+
+function _markPending(bucket, versions, cb) {
+    const options = { Bucket: bucket };
+    waterfall([
+        next => s3.getBucketReplication(options, (err, res) => {
+            if (err) {
+                log.error('error getting bucket replication', { error: err });
+                return next(err);
+            }
+            return next(null, res.ReplicationConfiguration);
+        }),
+        (repConfig, next) => mapLimit(versions, 10, (i, apply) => {
```

Comment:
10 may not be a good fit for everyone, so should be made configurable or a high-level exposed constant (also see my earlier comment on argument parsing).